internal/nodes/avi_vrf_translator.go

```diff
@@ -97,10 +97,17 @@ func (o *AviObjectGraph) BuildVRFGraph(key, vrfName, nodeName string, deleteFlag
 			diff := lenNewNodeRoutes - nodeStaticRouteDetails.Count
 
 			var staticRouteCopy []*models.StaticRoute
-			staticRouteCopy = append(staticRouteCopy, aviVrfNode.StaticRoutes[:startIndex]...)
+			if startIndex > -1 && startIndex <= len(aviVrfNode.StaticRoutes) {
+				staticRouteCopy = append(staticRouteCopy, aviVrfNode.StaticRoutes[:startIndex]...)
+			} else {
+				utils.AviLog.Warnf("Some StaticRoutes could not be updated because index is either -ve or out of bound")
+			}
 			staticRouteCopy = append(staticRouteCopy, nodeRoutes...)
-
-			staticRouteCopy = append(staticRouteCopy, aviVrfNode.StaticRoutes[startIndex+nodeStaticRouteDetails.Count:]...)
+			if (startIndex+nodeStaticRouteDetails.Count) > -1 && (startIndex+nodeStaticRouteDetails.Count) < len(aviVrfNode.StaticRoutes) {
+				staticRouteCopy = append(staticRouteCopy, aviVrfNode.StaticRoutes[startIndex+nodeStaticRouteDetails.Count:]...)
+			} else {
+				utils.AviLog.Warnf("Some StaticRoutes could not be updated because index is either -ve or out of bound")
+			}
 			aviVrfNode.StaticRoutes = staticRouteCopy
```

Comment:
can the staticRouteCopy be empty here after it goes into the else condition both the times? do we need some more handling or simply logs if staticRouteCopy is empty?

Reply:
It should not be empty as this code should add the latest static routes for the node to staticRouteCopy `staticRouteCopy = append(staticRouteCopy, nodeRoutes...)` but that will be all that it will have, so I also think we may need  to take care of this.

Reply:
@aaha97 and @DixitAakash , are you both suggesting, adding a length check here as well on staticRouteCopy and printing an error message

Reply:
The goal of this PR was to avoid any panic arising as a result of -ve index or IndexOutOfBound errors